File: Reformat Phone Number/string.py
def solution(number):
    number = number.replace(' ', ''). replace('-', '')
    ans, i, n = [], 0, len(number)
    while i < n:
        if n - i in [2, 4]:
            ans.append(number[i:i+2])
            i += 2
        else:
            ans.append(number[i:i+3])
            i += 3

    return '-'.join(ans)


if __name__ == "__main__":
    
    number = "1-23-45 6"
    solution(number)
    number = "123 4-567"
    solution(number)
    number = "123 4-5678"
    solution(number)
    number = "12"
    solution(number)
    number = "--17-5 229 35-39475 "
    solution(number)
    

File: Reformat Phone Number/string.cpp
#include <iostream>
#include <string>
using namespace std;

string solution(string number) {
    string num, ans;
    for (char c : number) {
        if (c != ' ' &&  c != '-') num += c;
    }

    int i = 0, n = num.size(); 
    while (i < n) {
        int step = n-i == 4 || n-i == 2 ? 2 : 3;
        ans += num.substr(i, step) + "-";
        i += step;
    }
    ans.pop_back();
    return ans;
}

int main() {
    string number = "1-23-45 6";
    cout << solution(number) << endl;
    number = "123 4-567";
    cout << solution(number) << endl;
    number = "123 4-5678";
    cout << solution(number) << endl;
}